sdk/python/feast/client.py

```diff
@@ -844,12 +843,15 @@ def ingest(
             produce = producer.produce
             flush = producer.flush
 
+            dataset_id = _generate_dataset_id(feature_set)
```

Comment:
I can imagine users really benefiting from being able to set this id. Can we also expose it as a field for them to optionally set?

Reply:
The use case we discussed previously was one where users could use their granularity as the dataset id, for example `20200301` as the dataset id.

Reply:
The danger is that it could then potentially not be unique, and we don't want to have to check for that :/

Reply:
> The danger is that it could then potentially not be unique, and we don't want to have to check for that :/

Well we would be pushing that responsibility to the producer. I am happy to start with this current implementation, but something tells me they want to have readable dataset ids.

Or are you more concerned about the stat generation?

Reply:
Alternatively a portion is unique and a portion can be manually set

[user or auto]-[uuid8]

Reply:
Yeah I understand the use case here. I think your latter suggestion makes sense